### PR TITLE
Add feed which imports domains and ipv4 mining pools from mining pool…

### PR DESCRIPTION
**Pull request by udgover**

### CODE REVIEW

**Comment by udgover** [comment]
It's related to minerstat feature request but relies on another source. Latest update from Minerstat feed is end of November 2023.

File: plugins/feeds/public/miningpoolstats.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        if response.status_code != 200:
+            logging.debug(f"Can't fetch pool page - code: {response.status_code}, reason: {response.reason}")
+            return
+        for data in response.json().get('data', []):
+            yield data
+        return
+
+    def _get_coin_names(self):
+        coin_names = set()
+        headers = {"User-Agent": self._USER_AGENT}
```
Comment (tomchop) [inline plugins/feeds/public/miningpoolstats.py:75]: Consider (re)using a requests Session object across the feed (will multiplex HTTP requests using the same UA and any cookies that are set)

File: plugins/feeds/public/miningpoolstats.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            return
+        response = requests.get(m.group(1), headers=headers)
+        if response.status_code != 200:
+            logging.debug(f"Can't fetch pool page - code: {response.status_code}, reason: {response.reason}")
+            return
+        for data in response.json().get('data', []):
+            yield data
+        return
+
+    def _get_coin_names(self):
```
Comment (tomchop) [inline plugins/feeds/public/miningpoolstats.py:73]: plz add type annotations

**Review by tomchop — CHANGES_REQUESTED** [review]

**Review by tomchop — APPROVED** [review]
one small nit, otherwise lgtm!

File: plugins/feeds/public/miningpoolstats.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        total = len(coin_names)
+        logging.info(f"Found {total} coin names")
+        count = 1
+        for coin_name in coin_names:
+            logging.info(f"Processing {coin_name} ({count}/{total})")
+            count += 1
+            tags = ["cryptomining", "cryptominer", "cryptomining-pool", coin_name]
+            for data in self._extract_pool_urls(coin_name):
+                url_value = data.get("url", "").strip()
+                url_obs = None
```
Comment (tomchop) [inline plugins/feeds/public/miningpoolstats.py:42]:
```suggestion
```

**Review by tomchop — CHANGES_REQUESTED** [review]

**Pull request merged**